kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,872 @@
+        ModelPart& r_surrogate_sub_model_part = rIgaModelPart.GetSubModelPart(surrogate_sub_model_part_name);
+
+        if (!rSkinModelPartInitial.HasProperties(0)) rSkinModelPartInitial.CreateNewProperties(0);
+        if (!rSkinModelPart.HasProperties(0)) rSkinModelPart.CreateNewProperties(0);
+        auto p_cond_prop_in = rSkinModelPartInitial.pGetProperties(0);
+        rSkinModelPartInitial.AddProperties(p_cond_prop_in);
+
+        array_1d<double, 2> knot_step_uv(2);
+        knot_step_uv[0] = std::abs(knotVectorU[std::ceil(knotVectorU.size()/2) +1]  - knotVectorU[std::ceil(knotVectorU.size()/2)] ) ;
+        knot_step_uv[1] = std::abs(knotVectorV[std::ceil(knotVectorV.size()/2) +1]  - knotVectorV[std::ceil(knotVectorV.size()/2)] ) ;
+
+        Vector mesh_sizes_uv(2);
+        mesh_sizes_uv[0] = knot_step_uv[0]; 
+        mesh_sizes_uv[1] = knot_step_uv[1];
+        auto& surrogate_model_part = rIgaModelPart.GetSubModelPart(surrogate_sub_model_part_name);
+        surrogate_model_part.GetProcessInfo().SetValue(MARKER_MESHES, mesh_sizes_uv);
+
+        array_1d<double, 2> starting_pos_uv;
+        starting_pos_uv[0] = knotVectorU[0];
+        starting_pos_uv[1] = knotVectorV[0];
+
+        Vector parameter_external_coordinates(4);
+        parameter_external_coordinates[0] = knotVectorU[0];
+        parameter_external_coordinates[1] = knotVectorV[0];
+        parameter_external_coordinates[2] = knotVectorU[knotVectorU.size()-1];
+        parameter_external_coordinates[3] = knotVectorV[knotVectorV.size()-1];
+        
+        surrogate_model_part.GetProcessInfo().SetValue(LOAD_MESHES, parameter_external_coordinates);
+
+        // Create the matrix of active/inactive knot spans, one for inner and one for outer loop
+    
+        std::vector<int> n_knot_spans_uv(2);
+        n_knot_spans_uv[0] = knotVectorU.size()-1; 
+        n_knot_spans_uv[1] = knotVectorV.size()-1;
+
+        std::vector<std::vector<std::vector<int>>> knot_spans_available;
+        knot_spans_available.reserve(numberOfLoops);
+
+        for (IndexType i = 0; i < numberOfLoops; ++i) {
+            std::vector<std::vector<int>> matrix; 
+            matrix.reserve(n_knot_spans_uv[1]);
+            for (int j = 0; j <= n_knot_spans_uv[1]-1; ++j) {
+                std::vector<int> row(n_knot_spans_uv[0]); 
+                matrix.push_back(row); 
+            }
+            knot_spans_available.push_back(matrix);
+        }
+        
+        // Optimized Snake -> for inner loops
+        int id_matrix_knot_spans_available = 0;
+        IndexType id_first_node;
+        bool new_inner_loop = true;
+
+        
+        KRATOS_INFO_IF("::[SnakeSbmUtilities]::", r_echo_level > 0 && is_inner) << "Inner :: Starting SnakeStep" << std::endl;
+        KRATOS_INFO_IF("::[SnakeSbmUtilities]::", r_echo_level > 0 && !is_inner) << "Outer :: Starting SnakeStep" << std::endl;
+                
+        if (rSkinModelPartInitial.Conditions().size() > 0) {
```

Comment:
```suggestion
        if (rSkinModelPartInitial.NumberOfConditions() > 0) {
```

Reply:
For the sake of the future yourselves, I'd throw a warning if there are no conditions in the provided skin model part.